src/proxy_app/telemetry/logger.py

```
@@ -0,0 +1,267 @@
+                pass  # no running loop yet; will start on first async hook
+
+    # --- pre/post call hooks (sync, fast) ---
+    def log_pre_api_call(self, model, messages, kwargs):
+        try:
+            rid = kwargs.get("litellm_call_id") or str(id(kwargs))
+            self._start_times[rid] = _now_ms()
+            stream = kwargs.get("stream", False)
+            if stream and rid not in self._first_token_times:
+                self._first_token_times[rid] = 0.0
+        except Exception:
+            pass  # never raise in callback
```

Ruff (0.15.17)

[error] `try`-`except`-`pass` detected, consider logging the exception

(S110)